### DISCUSSION TITLE
Browser Intelligent Automation

### DISCUSSION THREAD

**Discussion by wkjw** [adamlui/ai-web-extensions#182] [Ideas]
Is it possible to automatically identify the content in the browser page, then submit it to GPT so that it can automatically get reasonable operations, and then expand the function of automatically operating the browser content based on the reply content to achieve intelligent automatic operation of the web page?

**Comment by adamlui** [comment]
Can you give an example?

**Reply by KernTwo** [reply]
I am trying to work on something similar. May I suggest a format that looks like InDesign, where you can change content via links on a drive—within an EPUB or PDF—for editing (PDF is more of a hassle, though, as it is based on so many different PDF producers and settings).
The key feature would be something TOC-based, and AI access to change names or rearrange numbered chapters, subsections... 

There could be a function for activating parts of a document (from File Explorer: folders, text documents, images); this would require tools to select text, similar to MS Word’s "Find and Replace" feature, selecttion tool, or page number or name definded; from - to, odd, even, current. Variable data functionality would also be useful. Imagine a lexicon of every topic, requiring specific styles for various subjects. Living hotfolders.

Another use could be a "Sandbox" where you can dump files and everything else, then organize and puzzle them into shape. Like a collage of work intended for reference, inspiration - mixing styles (deep and wide reasoning, or less processor intensive "holding" a more limitied frame of subjects, scientific fields or philosophical "thinking". So if you want to make Trump sound like Musk in a document, and start a casino empire on the moon... whatever. 

More "dreaming" up what would be nice to have.  

If PDF based, Enfocus Pitstop is a prime example for tool that could be needed to select and change elements within. 

Nice if it had a feature that could handle metadata, on detail level or general tags, like icc color profiles, fonts ... 

 Maybe it is best to just go web based, and get the TOC coded - linking to source folders. In theory you could change entire pages inside the document - and have comments and prompts already on the site. 
 
 A key requirement would have to be "update" new content from the source-folder. 
 
 Updatebased, instead of dialogue based. 
 
 A very big feature, it would allow for various styles of working with long text documents. Books, essays, journals or magazines... 
 
File explorer, notepad, html, css, json..., Indesign/epub/adobe bridge. 

It would be the perfect addon to take notes with, and store where you need it for various projects - at once

Too much?

**Comment by wkjw** [comment]
Of course, for example, if I open a webpage with many logical questions in it, I hope to have a plug-in that can read all the questions on the page, then identify the best answer, perform automated operations, and then submit

**Comment by adamlui** [comment]
Can you give a detailed example, with specific questions that appear, then what query is sent to GPT
